Source/Core/Common/RingBuffer.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <algorithm>
+#include <atomic>
+#include <cstring>
+#include <vector>
+
+#include "Common/MathUtil.h"
+
+// Simple Ring Buffer class.
+// Since we use bitwise & for circularity for speed, m_max_size will be adjusted to power of 2.
+// Only implemented bulk data writing.
+// Add functions as necessary.
+template <class T>
```

Comment:
Mention that this only works with POD types, and potentially even add a static_assert for it.